### PR TITLE
Fix dissect trim panics from DELETE (\u007f) character

### LINKED ISSUE #30657: Dissect trim panics from unicode DELETE (\u007f) char
We hit a panic with parsing logs using dissect with trim when a line had a DELETE (127)(\u007f) character at the end of the line.

- Version: 7.17.0 (affects 7.9 and later)
- Operating System: Linux
- Steps to Reproduce: Test below

```
panic: runtime error: index out of range [127] with length 127

goroutine 52 [running]:
github.com/elastic/beats/v7/libbeat/processors/dissect.asciiTrimmer.Trim(...)
	/go/src/github.com/elastic/beats/libbeat/processors/dissect/trim.go:66
github.com/elastic/beats/v7/libbeat/processors/dissect.(*Dissector).Dissect(0xc000a0da40, 0xc001517d10, 0xe5, 0x108528980, 0xc0015b1060, 0x0)
	/go/src/github.com/elastic/beats/libbeat/processors/dissect/dissect.go:73 +0x104
github.com/elastic/beats/v7/libbeat/processors/dissect.(*processor).Run(0xc0005b5280, 0xc0015c81c0, 0x107d95b74, 0x1e, 0xc000b8bad0)
	/go/src/github.com/elastic/beats/libbeat/processors/dissect/processor.go:91 +0x3e5
```

Test to trigger the panic:

```
func TestTrimmer(t *testing.T) {
	for _, test := range []struct {
		name, cutset    string
		left, right     bool
		input, expected string
	}{
		{
			name:     "trim both with unicode",
			cutset:   " ",
			left:     true,
			right:    true,
			input:    "  hello world! \u007f ",
			expected: "hello world! \u007f",
		},
	} {
		t.Run(test.name, func(t *testing.T) {
			trimmer, err := newTrimmer(test.cutset, test.left, test.right)
			if !assert.NoError(t, err) {
				return
			}
			start, end := trimmer.Trim(test.input, 0, len(test.input))
			output := test.input[start:end]
			assert.Equal(t, test.expected, output)
		})
	}
}
```

### PR DESCRIPTION
**Pull request by montaguethomas**
## What does this PR do?

Fixes an index out of range panic with `asciiTrimmer.Trim()`.

## Why is it important?

Should a string begin or end with the DELETE (\u007f)(127) ASCII character (with or without a leading/trailing space), `asciiTrimmer.Trim()` will throw a panic.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~[ ] I have made corresponding changes to the documentation~
- ~[ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Run the updated tests without the fix in trim.go.

## Related issues

- Closes #30657 

## Logs

```
panic: runtime error: index out of range [127] with length 127

goroutine 52 [running]:
github.com/elastic/beats/v7/libbeat/processors/dissect.asciiTrimmer.Trim(...)
	/go/src/github.com/elastic/beats/libbeat/processors/dissect/trim.go:66
github.com/elastic/beats/v7/libbeat/processors/dissect.(*Dissector).Dissect(0xc000a0da40, 0xc001517d10, 0xe5, 0x108528980, 0xc0015b1060, 0x0)
	/go/src/github.com/elastic/beats/libbeat/processors/dissect/dissect.go:73 +0x104
github.com/elastic/beats/v7/libbeat/processors/dissect.(*processor).Run(0xc0005b5280, 0xc0015c81c0, 0x107d95b74, 0x1e, 0xc000b8bad0)
	/go/src/github.com/elastic/beats/libbeat/processors/dissect/processor.go:91 +0x3e5
```

### CODE REVIEW

**Comment by cmacknz** [comment]
@adriansr you were the original author here, can you review this fix?

**Comment by cmacknz** [comment]
@adriansr you were the original author here, can you review this fix?

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -43,6 +43,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...main[Check the HEAD dif
 - Update docker/distribution dependency library to fix a security issues concerning OCI Manifest Type Confusion Issue. {pull}30462[30462]
 - Fixes Beats crashing when glibc >= 2.35 is used {issue}30576[30576]
 - Log errors when parsing and applying config blocks and if the input is disabled. {pull}30534[30534]
+- Fix dissect trim panics from DELETE (127)(\u007f) character {issue}30657[30657] {pull}30658[30658]
 - Wildcard fields no longer have a default ignore_above setting of 1024. {issue}30096[30096] {pull}30668[30668]
 - Ignore bugfix version when running version compatibility check against Elasticsearch. {pull}30746[30746]
```
Comment (adriansr) [inline CHANGELOG.next.asciidoc:48]:
```suggestion
- Log errors when parsing and applying config blocks and if the input is disabled. {pull}30534[30534]
- Wildcard fields no longer have a default ignore_above setting of 1024. {issue}30096[30096] {pull}30668[30668]
- Ignore bugfix version when running version compatibility check against Elasticsearch. {pull}30746[30746]
- Fix dissect trim panics from DELETE (127)(\u007f) character {issue}30657[30657] {pull}30658[30658]
```

Comment (montaguethomas) [reply CHANGELOG.next.asciidoc]: haha! i got tired of of the merge conflicts after the second one. left the ordering of entries by the number of the PR.

**Pull request merged**